spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatch.java

```diff
@@ -65,18 +63,24 @@ public InputPartition[] planInputPartitions() {
     Broadcast<Table> tableBroadcast = sparkContext.broadcast(SerializableTable.copyOf(table));
     String expectedSchemaString = SchemaParser.toJson(expectedSchema);
 
-    InputPartition[] readTasks = new InputPartition[tasks.size()];
+    InputPartition[] readTasks = new InputPartition[tasks().size()];
 
     Tasks.range(readTasks.length)
         .stopOnFailure()
         .executeWith(localityEnabled ? ThreadPools.getWorkerPool() : null)
         .run(index -> readTasks[index] = new ReadTask(
-            tasks.get(index), tableBroadcast, expectedSchemaString,
+            tasks().get(index), tableBroadcast, expectedSchemaString,
```

Comment:
[minor] task() is synchronized, and we might be accessing here with a pool, should remove synchronized from function def and make it like : 

```
private volatile List<CombinedScanTask> tasks = null

protected synchronized List<CombinedScanTask> tasks() {
    if (tasks == null) {
      synchronized(this) {
        if (tasks == null) {
            task = // assign stuff to task
         }
       }
    }
    return tasks;
  }
```

I think now we might also want to make `SparkFilesScan#task()` handle multi-threading.

Reply:
It looks like we weren't synchronizing before in 0.13, but I can add that check if you think it's needed

Reply:
I see, as per my understanding, was thinking, we would be requiring it, as we have a possibility to call tasks() via mt (as we were in a thread pool), Though I see now in L66 above we already would have called tasks() which  would have populated this `task` anyhow, since we have made the function itself [synchronized](https://github.com/apache/iceberg/blob/master/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkCopyOnWriteScan.java#L134), each of this access will now be sequential even for the case we have task as not null.

All in all would say I don't have a strong reason for it, would say it's just something that caught my eye :) !